packages/core/lib/console-child.js

```diff
@@ -0,0 +1,45 @@
+const Command = require("../lib/command");
+const TruffleError = require("@truffle/error");
+const Config = require("@truffle/config");
+const Web3 = require("web3");
+
+const inputStrings = process.argv[2];
+const network = process.argv[3];
```

Comment:
Can we merge these two? i.e., if the user runs `debug --network mainnet` from inside `truffle console --network development`, then I would expect the `debug` command to connect to mainnet. This implementation suggests that behavior might be undefined (or favor the network specified by `process.argv[3]`)

Seems like we shouldn't pass `network` separately as a positional argument, and instead pass it as a named `--network` option. Two options come to mind for how to do that:

1. Add logic to the parent process to merge `--network=<network-name>` into `inputStrings`, possibly by decoding `inputStrings` into a mapping + then re-encoding with `{ network }` merged in.

2. Use a ` -- ` separator between parent-process-provided options and user-provided options, e.g.:
  ```
  truffle(develop)> debug <tx-hash> --fetch-external
     ... would run command:
     $ console-child.js --network=develop -- <tx-hash> --fetch-external
  ```

Reply:
Hi @gnidan!  https://github.com/trufflesuite/truffle/pull/3296 addresses this in case you do want to get rid of the positional argument. Wanted to save you a little time!

Reply:
Thanks so much @fainashalts, thanks for knocking that out!

That PR looks great! But why is it failing ☹️ ? (Re-running now)